### ISSUE TITLE
Installation issues with Python 3.9 

### ISSUE THREAD

**Issue by aegerton** [casact/chainladder-python#122]
There appears to be an issue installing on Python 3.9 due to ```numba``` installation issues as detailed in the following Issue;-
```
https://github.com/numba/numba/issues/6345#issue-718596580
```
Is there a workaround available, please?

**Comment by jbogaardt** [comment]
Hi @aegerton ,  Thankd for reporting as I wasn't aware of this.  `numba` is a hard requirement by way of the `sparse` package.  `chainladder` doesn't have too many dependencies, but the ones it has are major ones.

- `pandas`
- `scikit-learn`
- `sparse`
- `dill`
- `xlcompose` (deprecated)

Most of these packages have professional development teams supporting them, numba included. So if they don't have a work-around, I doubt I'd be skilled enough to find one - specifically a workaround for using chainladder on python>=3.9.  `conda` environments help and if I create one with `chainladder` as the only requirement, it gives me `python==3.8.5` if you're okay with a downgrade.

#121 is another recent chainladder issue due to upgrading to the latest in other dependencies.  I think what I need to do with this issue is to create a "bleeding edge" CI/CD pipeline that tests against the latest releases of all dependencies to at least be aware of these conflicts when they arise.

**Comment by jbogaardt** [comment]
Oh, it's already in, which means it should be in the next release.

_Originally posted by @hameerabbasi in https://github.com/pydata/sparse/issues/426#issuecomment-762042578_

It looks like this should resolve once the `sparse` package does its next release.  It looks like their workaround is:

Sparse now installs for python 3.9 if the exact following version is used as a dependency:
`numba==0.53.0rc1.post1`. But a regular `pip install sparse` does not work unless `numba==0.53.0rc1.post1` is already installed.

_Originally posted by @JackWetherell in https://github.com/pydata/sparse/issues/426#issuecomment-777428853_

**Comment by aegerton** [comment]
Thanks @jbogaardt for the usual prompt and helpful response.  It is greatly appreciated.

I'll keep an eye open for the ```sparse``` release and try again then.  In the meantime, I have it working with ```Python 3.8.6``` after downgrading.  

Is it worth adding a note about dependency issues with ```3.9``` to the ```README```, please?

**Comment by jbogaardt** [comment]
> Is it worth adding a note about dependency issues with `3.9` to the `README`, please?

Yes, of course.

**Comment by hameerabbasi** [comment]
> It looks like this should resolve once the `sparse` package does its next release. It looks like their workaround is:

Small correction, `sparse` doesn't need a new release, `numba` does. `sparse` should magically "just work".

**Comment by jbogaardt** [comment]
Thank you @hameerabbasi.  You do fantastic work on `sparse`!

**Comment by JackWetherell** [comment]
I will leave a comment here once the final numba/llvmlite release is published, and the suggested workaround from my sparse issue is no longer needed.

**Comment by aegerton** [comment]
> > Is it worth adding a note about dependency issues with `3.9` to the `README`, please?
> 
> Yes, of course.

Thanks again @jbogaardt.  Saw the change.  Congratulations on the release of ```0.8.0```

**Comment by JackWetherell** [comment]
The numba/llvmlite release is published. The above workaround is no longer needed. sparse now installs with python 3.9.1. 
The current versions of the relevant dependencies are:
```
sparse==0.11.2
numba==0.53.0
llvmlite==0.36.0
```

**Comment by jbogaardt** [comment]
Thank you @JackWetherell for letting us know. I'll give it a try.

**Comment by jbogaardt** [comment]
Resolved by #139
